api/src/main/java/ai/djl/nn/AbstractSymbolBlock.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.nn;
+
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.training.ParameterStore;
+import ai.djl.training.initializer.Initializer;
+import ai.djl.util.PairList;
+import ai.djl.util.Resource;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+
+/**
+ * {@code AbstractSymbolBlock} is an abstract implementation of {@link SymbolBlock}.
+ *
+ * @param <T> the resource that SymbolBlock holds
+ */
+public abstract class AbstractSymbolBlock<T> extends Resource<T> implements SymbolBlock {
+
+    protected AbstractSymbolBlock(T handle) {
+        super(handle);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public abstract NDList forward(
+            ParameterStore parameterStore,
+            NDList inputs,
+            boolean training,
+            PairList<String, Object> params);
+
+    /** {@inheritDoc} */
+    @Override
+    public void removeLastBlock() {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void setInitializer(Initializer initializer) {
```

Comment:
SymbolBlock might be used as partial of a Block for training, we can assume a SymbolBlock is initialized already instead of failing initialize